paperqa/agents/tools.py

```diff
@@ -76,7 +69,11 @@ def status(self) -> str:
 
     def record_action(self, action: ToolRequestMessage) -> None:
         self.session.add_tokens(action)
-        self.tool_history.append([tc.function.name for tc in action.tool_calls])
+        self.session.tool_history.append([tc.function.name for tc in action.tool_calls])
+
+    def query_tool_history(self, tool_name: str) -> list[int]:
```

Comment:
Also fwiw, I can see what you coded making sense too, just it loses a bit of information (it could be "called x times at steps a, b, c", but as is it's "called at least once at steps a, b, c")

Reply:
The way it's implemented here makes more sense to me, it's the step number that a tool was called-- we don't have another use for this, I thought it could be something useful down the line. 

Since that's probably YAGNI territory 😄  -- I can just turn it into a straight up boolean because we likely don't need anything else. I'm just checking if it's in the history anywhere.

Reply:
Ha you went the opposite direction, I see and YAGNIIIII